editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -1222,6 +1225,119 @@ void SpriteFramesEditor::_animation_duplicate() {
+	undo_redo->add_undo_method(frames.ptr(), "set_animation_speed", edited_anim, frames->get_animation_speed(edited_anim));
+	undo_redo->add_undo_method(frames.ptr(), "set_animation_loop", edited_anim, frames->get_animation_loop(edited_anim));
+	for (int i = 0; i < frame_count; i++) {
+		Ref<Texture2D> texture = clipboard_anim->frames[i].texture;
+		float duration = clipboard_anim->frames[i].duration;
+		undo_redo->add_undo_method(frames.ptr(), "add_frame", edited_anim, texture, duration);
+	}
+	undo_redo->add_do_method(this, "_select_animation", new_edited);
+	undo_redo->add_undo_method(this, "_select_animation", edited_anim);
+	undo_redo->add_do_method(this, "_update_library");
+	undo_redo->add_undo_method(this, "_update_library");
+	undo_redo->commit_action();
+}
+
+void SpriteFramesEditor::_animation_copy() {
+	Ref<ClipboardAnimation> clipboard_anim = memnew(ClipboardAnimation);
+	clipboard_anim->animation_name = edited_anim;
+	clipboard_anim->animation_speed = frames->get_animation_speed(edited_anim);
+	clipboard_anim->animation_loop = frames->get_animation_loop(edited_anim);
+	clipboard_anim->is_copy = true;
+
+	int frame_count = frames->get_frame_count(edited_anim);
+	for (int i = 0; i < frame_count; ++i) {
+		ClipboardSpriteFrames::Frame frame;
+		frame.texture = frames->get_frame_texture(edited_anim, i);
+		frame.duration = frames->get_frame_duration(edited_anim, i);
+		clipboard_anim->frames.push_back(frame);
+	}
+
+	EditorSettings::get_singleton()->set_resource_clipboard(clipboard_anim);
+}
+void SpriteFramesEditor::_animation_paste() {
+	Ref<ClipboardAnimation> clipboard_anim = EditorSettings::get_singleton()->get_resource_clipboard();
+	if (!clipboard_anim.is_valid()) {
+		return;
+	}
+
+	if (updating) {
+		return;
+	}
+
+	if (!clipboard_anim->is_copy) {
+		EditorSettings::get_singleton()->set_resource_clipboard(nullptr);
+	}
+
+	int counter = 1;
+	String new_name = edited_anim;
+	PackedStringArray name_component = new_name.rsplit("_", true, 1);
+	String base_name = name_component[0];
+	if (name_component.size() > 1 && name_component[1].is_valid_int() && name_component[1].to_int() >= 0) {
+		counter = name_component[1].to_int();
+	}
+	new_name = base_name + "_" + itos(counter);
+	while (frames->has_animation(new_name)) {
+		counter++;
+		new_name = base_name + "_" + itos(counter);
+	}
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Paste Animation"), UndoRedo::MERGE_DISABLE, EditorNode::get_singleton()->get_edited_scene());
+	undo_redo->add_do_method(frames.ptr(), "add_animation", new_name);
+	undo_redo->add_undo_method(frames.ptr(), "remove_animation", new_name);
+	undo_redo->add_do_method(frames.ptr(), "set_animation_speed", new_name, clipboard_anim->animation_speed);
+	undo_redo->add_do_method(frames.ptr(), "set_animation_loop", new_name, clipboard_anim->animation_loop);
+
+	for (const auto &frame : clipboard_anim->frames) {
```

Comment:
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)